scipy/stats/tests/mpmath_distribution.py

```diff
@@ -0,0 +1,169 @@
+import numpy as np
+from mpmath import mp
+
+mp.dps = 100  # default in case the user forgets to set it
+
+class Distribution:
+    # Minimalist distribution infrastructure for generating reference values
+    # of distribution functions. No edge case handling.
+
+    @staticmethod
+    def _make_mpf_array(x):
+        shape = np.shape(x)
+        x = np.asarray(x, dtype=np.float64).ravel()
+        return np.asarray([mp.mpf(xi) for xi in x]).reshape(shape)[()]
+
+    def __init__(self, **kwargs):
+        self._params = {key:self._make_mpf_array(val)
+                        for key, val in kwargs.items()}
+
+    def _pdf(self, x):
+        raise NotImplementedError("_pdf must be overridden.")
+
+    def _cdf(self, x, **kwargs):
+        a, _ = self._support(**kwargs)
+        return mp.quad(lambda x: self._pdf(x, **kwargs), (a, x))
+
+    def _sf(self, x, **kwargs):
+        _, b = self._support(**kwargs)
+        return mp.quad(lambda x: self._pdf(x, **kwargs), (x, b))
+
+    def _logpdf(self, x, **kwargs):
+        return mp.log(self._pdf(x, **kwargs))
+
+    def _logcdf(self, x, **kwargs):
+        return mp.log(self._cdf(x, **kwargs))
+
+    def _logsf(self, x, **kwargs):
+        return mp.log(self._sf(x, **kwargs))
+
+    def _support(self, **kwargs):
+        return -mp.inf, mp.inf
+
+    def _entropy(self, **kwargs):
+        def integrand(x):
+            logpdf = self._logpdf(x, **kwargs)
+            pdf = mp.exp(logpdf)
+            return  -pdf*logpdf
+
+        a, b = self._support(**kwargs)
+        return mp.quad(integrand, (a, b))
+
+    def _mean(self, **kwargs):
```

Comment:
I'm thinking about adding memoization to speed things up, but I'm going to keep it simple for now.